tests/deepsparse/transformers/pipelines/test_chat.py

```diff
@@ -12,34 +12,157 @@
+            else 1,
+        )
+
+        # make sure information does not leak between sessions
+        self._test_composition_same_session_ids(
+            prompt_1,
+            prompt_2,
+            num_generated_tokens,
+            pipeline,
+            session_id_1="test_1",
+            session_id_2="test_2",
+        )
+
+        self._test_composition_same_session_ids(
+            prompt_1,
+            prompt_2,
+            num_generated_tokens,
+            pipeline,
+            session_id_1="test_3",
+            session_id_2="test_4",
+        )
+
+    def _test_composition_same_session_ids(
+        self,
+        prompt_1,
+        prompt_2,
+        num_generated_tokens,
+        pipeline,
+        session_id_1,
+        session_id_2,
+    ):
+
+        tokenizer = pipeline.tokenizer
+        config = GenerationConfig(
+            output_scores=True, max_length=num_generated_tokens, top_k=0, top_p=0.0
+        )
+
+        # make sure that running two prompts one after another
+        # is identical to running a composition of those prompts
+        out_1_ = pipeline(
+            sequences=prompt_1,
+            force_max_tokens=True,
+            session_ids=session_id_1,
+            generation_config=config,
+            include_prompt_logits=True,
         )
-        output_2 = chat_pipeline(
-            prompt="second", generation_config=dict(max_new_tokens=1)
+        prompt_1_ = out_1_.generations[0].text
+        out_1 = pipeline(
+            sequences=prompt_2,
+            force_max_tokens=True,
+            session_ids=session_id_1,
+            generation_config=config,
+            include_prompt_logits=True,
         )
-    # assert inferences in the same context share a session id
-    assert output_1.session_ids == output_2.session_ids
+        cache_state_1 = pipeline.storage_kv_cache.get(session_id_1).cached_inputs[
+            "past_key_values.0.key"
+        ]
 
-    # test that follow-up inference has a different session id
-    output_3 = chat_pipeline(prompt="third", generation_config=dict(max_new_tokens=1))
-    assert output_3.session_ids != output_1.session_ids
+        prompt_composition = tokenizer.decode(
+            tokenizer(prompt_1).input_ids
+            + tokenizer(prompt_1_).input_ids
+            + tokenizer(prompt_2).input_ids,
+            skip_special_tokens=True,
+        )
+        out_2 = pipeline(
+            sequences=prompt_composition,
+            session_ids=session_id_2,
+            generation_config=config,
+            include_prompt_logits=True,
+        )
+        cache_state_2 = pipeline.storage_kv_cache.get(session_id_2).cached_inputs[
+            "past_key_values.0.key"
+        ]
+        if cache_state_1.shape[0]:
+            # if cache state is not empty, i.e. we are managing kv cache
+            # externally, make sure that the cache state is the same
+            np.allclose(cache_state_1, cache_state_2, atol=self.precision)
```

Comment:
what is cache_state_n and what does it mean to be +/- self.precision?  If I make session1 and a duplicate of session 1 above, will the cache_state_n of those be the same?

The only difference if stateful between sess1 and sess2  is the number of times pipeline was called w the respective session_id right? is this causing the difference?